elrond-wasm-derive/src/arg_regular.rs

```diff
@@ -5,71 +5,69 @@ use super::util::*;
 
 pub fn generate_load_single_arg(arg: &MethodArg, arg_index_expr: &proc_macro2::TokenStream) -> proc_macro2::TokenStream {
     let arg_ty = &arg.ty;
-    let arg_name_literal = pat_literal(&arg.pat);
+    let arg_name_expr = arg_id_literal(&arg.pat);
     match &arg.ty {
         syn::Type::Reference(type_reference) => {
             if type_reference.mutability.is_some() {
                 panic!("Mutable references not supported as contract method arguments");
             }
             let referenced_type = &*type_reference.elem;
             quote! {
-                & elrond_wasm::load_single_arg::<T, BigInt, BigUint, #referenced_type>(&self.api, #arg_index_expr, #arg_name_literal)
+                & elrond_wasm::load_single_arg::<T, BigInt, BigUint, #referenced_type>(self.api.clone(), #arg_index_expr, #arg_name_expr)
             }
         },
         _ => {
             quote! {
-                elrond_wasm::load_single_arg::<T, BigInt, BigUint, #arg_ty>(&self.api, #arg_index_expr, #arg_name_literal)
+                elrond_wasm::load_single_arg::<T, BigInt, BigUint, #arg_ty>(self.api.clone(), #arg_index_expr, #arg_name_expr)
             }
         },
     }
 }
 
 pub fn generate_load_dyn_arg(arg: &MethodArg,
-        loader_expr: &proc_macro2::TokenStream,
-        err_handler_expr: &proc_macro2::TokenStream) -> proc_macro2::TokenStream {
+        loader_expr: &proc_macro2::TokenStream) -> proc_macro2::TokenStream {
```

Comment:
Can you rename proc_macro2?

Reply:
Not mine, it is part of the macro framework :)

Reply:
Wish I could :))